packages/common/src/core/types/IHubEntityBase.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Base properties for Hub Entities
+ * This is a subset of `IItem`, that can apply to
+ * models that are not backed by items.
+ */
+export interface IHubEntityBase {
+  /**
+   * Id of the entity as a string
+   */
+  readonly id?: string;
+  /**
+   * Name of the Entity
+   * For Entities backed by items, this would be the title
+   */
+  name: string;
+  /**
+   * Simple summary of the Entity.
+   * For items, this is the `snippet` property.
+   */
+  summary?: string;
+  /**
+   * Date the entity was created
+   */
+  createdDate: Date;
+  /**
+   * Source of the creation date
+   */
+  createdDateSource: "record" | "item.created" | "item.metadata.created_date";
+  /**
+   * Date the entity was last updated
+   */
+  updatedDate: Date;
+  /**
+   * Source of the updated date
+   */
+  updatedDateSource: "record" | "item.modified" | "item.metadata.modified_date";
```

Comment:
I also don't have a strong preference... and since `created` is a value on an item, making `created` be a `IDateInfo` object may be confusing. I'll leave as-is for now